pkg/controller/reconciler/deployment_reconciler.go

```diff
@@ -250,12 +250,14 @@ func (r *DeploymentReconciler) reconcileUpdate(ctx context.Context, deployment *
 	}
 
 	environmentName := "default"
-	if annotations.Configuration.Environment != "" {
+	// What if Configuration is nil?
```

Comment:
I don't mind the extra check, but this method will not be called when the configuration is nil.